### PR TITLE
[oneDNN ] disabling more ops caching

### PR DESCRIPTION
**Pull request by jczaja**
### PR types
Bug fixes

### PR changes
OPs 

### Describe
Recently oneDNN introduced caching of its elements: primitives and primitives descriptors so we can disable partially our own caching. This PR disabled caching of LRN, Transpose, Layer Norm and Sum oneDNN ops

### CODE REVIEW

File: paddle/fluid/operators/mkldnn/sum_mkldnn_op.cc [hunk 1]
```diff
@@ -45,82 +45,63 @@ using paddle::platform::MKLDNNDeviceContext;
...
-    const std::string key_pd = this->key_ + "@fwd_pd";
-    this->fwd_pd_ = std::static_pointer_cast<dnnl::sum::primitive_desc>(
-        this->dev_ctx_.GetBlob(key_pd));
-    if (this->fwd_pd_ == nullptr) {
-      this->fwd_pd_.reset(new dnnl::sum::primitive_desc(dst_md, scales, srcs_md,
-                                                        this->engine_));
-      this->dev_ctx_.SetBlob(key_pd, this->fwd_pd_);
-    }
+    this->fwd_pd_.reset(
+        new dnnl::sum::primitive_desc(dst_md, scales, srcs_md, this->engine_));
```
Comment (jakpiase) [inline paddle/fluid/operators/mkldnn/sum_mkldnn_op.cc:85]: You are using "new" here. Is it later destroyed anywhere?

Comment (jczaja) [reply paddle/fluid/operators/mkldnn/sum_mkldnn_op.cc:85]: this->fwd_pd_ is a shared pointer so when reference count will get to zero then deallocation of  of object will happen . So we do not have to worry about memory leaks.

Comment (jakpiase) [reply paddle/fluid/operators/mkldnn/sum_mkldnn_op.cc:85]: Ok, thanks. I didn't see that it is a smart pointer

**Review by jakpiase — APPROVED** [review]
LGTM

**Review by lidanqing-vv — APPROVED** [review]
LGTM.

**Pull request merged**